src/modules/did.js

```diff
@@ -0,0 +1,120 @@
+const ChluDID = require('chlu-did/src')
+const { getDigestFromMultihash } = require('../utils/ipfs')
+
+class ChluIPFSDID {
+
+    static isDIDID(didId) {
+        return typeof didId === 'string' && didId.indexOf('did:') === 0
+    }
+
+    constructor(chluIpfs) {
+        this.chluIpfs = chluIpfs 
+        // ChluDID instance
+        this.chluDID = new ChluDID()
+        // Well Known DIDs
+        // used for hardcoded DIDs in tests
+        // and will also be used for stuff like official Chlu trusted DIDs
+        this.wellKnownDIDs = {}
+        this.didId = null
+        this.publicDidDocument = null
+        this.privateKeyBase58 = null
+    }
+
+    async start() {
+        if (!this.isPresent()) {
+            // Generate a DID & Publish
+            await this.generate();
+            await this.chluIpfs.persistence.persistData();
+        }
+    }
+
+    async generate() {
+        const did = await this.chluDID.generateDID()
+        return await this.import(did)
+    }
+
+    async import(did, publish = true, waitForReplication = false) {
+        this.publicDidDocument = did.publicDidDocument
+        this.didId = this.publicDidDocument.id
+        this.privateKeyBase58 = did.privateKeyBase58
+        await this.chluIpfs.persistence.persistData()
+        if (publish) await this.publish(null, waitForReplication)
+    }
+
+    export() {
```

Comment:
This `export` just returns the in memory copy of the loaded DID for the user.

If we move it to ChluDID it means that ChluDID has to handle persistence of the DID in the browser and in node.js... I think we should have persistence handled in only one place like we do now.

This export function is just a quick reliable way to get the object representing the whole DID, I think it is only used in the persistence module, to save it to file/localstorage, and in case you need a copy for any other reason, for example in the wallet to keep it in the redux store.

ChluDID does not keep any DIDs, you have to pass it for every operation.

Let me know what you think 😄

Reply:
I agree this did.js module should handle persistence. I see what you are doing now. I would have done it slightly differently by making `did` an 'instance variable' and then export would really just be a getter and you won't have to disassemble the did into various instance variables.

The benefit of that is that then if you want to provide more functionality for dids it goes into chlu-did and you "call a method on the object". Just trying to keep things encapsulated. By breaking up the did received from chlu-did we break encapsulation.

Reply:
I see, you'd like the DID to be an object with methods instead of just some data, right?

Reply:
Well, chlu-did already provides some responsibilities like sign and verify, and if we keep things encapsulated, then in the future it is easy to add things to the same "class".

I must be sounding so old school using all the OO lingo.

Reply:
Haha, yeah OOP is not cool anymore apparently.

But I agree completely... now verify and sign work using the functional way and the did is passed as a parameter, like this

```javascript
lib.sign(myDid, ...)
```

But I agree that for DIDs OOP might be better:

```javascript
myDid.sign(...)
```

Opened an issue to track this: https://github.com/ChluNetwork/chlu-did/issues/3